cloud-init-configs/auto-ssh.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+USER="$1"
+VM_NAME="$2"
+
+ip_addr=$(virsh domifaddr "$VM_NAME" | \
+          grep -oE '[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+' | \
+          awk '{print $1}'\
+        )
+
+echo "$VM_NAME : $ip_addr"
+
+ssh "$USER"@"$ip_addr"
```

Comment:
All those variables could be an issue. Instead what if you do it "functionally"? Something like:

`ssh -t -i ./id_ed25519 -o StrictHostKeyChecking=no $USER@$(virsh domifaddr VMNAME | awk -F'[ /]+' '{if (NR>2) print $5}')`

Reply:
I tried this and it worked. I removed the -i <file> because I don't believe I need that. 

But what is the NR>2? NR is a built in from awk that counts the number of records? So basically what it is doing is looking for the section from the domifaddr output that has more than 2 records? is that how were finding the ip in this case?

Reply:
That's a different thing. What I'm pointing out is that you don't need to set variables and then call them you can just do everything in one line with the subshell embedded with the $args in the ssh command.

Reply:
`ssh $1@$(virsh domifaddr $2| grep -oE '[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+' | awk '{print $1}')`

Reply:
You can add a comment to help explain what's going on instead of formatting your code like that.